### PR TITLE
nuttx: add get_current_dir_name implementation

### PR DESCRIPTION
**Pull request by extinguish**
## Summary
add `get_current_dir_name` API implementation

## Impact
has no impact on other module

## Testing
has passed ostest

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No. The PR description lacks crucial details required by the NuttX guidelines.  While it mentions the added API, it fails to provide context or justification.  Here's why it's insufficient and what needs improvement:

* **Insufficient Summary:** "add `get_current_dir_name` API implementation" is too brief.  Explain *why* this API is needed. What problem does it solve? What use cases does it address? Which part of the codebase is affected (e.g., VFS, libc)? How does the implementation work (e.g., does it traverse the path, use a cached value, etc.)?

* **Insufficient Impact Assessment:** "has no impact on other module" is likely untrue.  Almost any change has *some* impact.  Consider:
    * **User Impact:** Will applications need to be modified to use this new API?
    * **Build Impact:**  Are any new dependencies introduced?  Do any build flags need to be set?
    * **Hardware Impact:**  Does it rely on specific hardware features?
    * **Documentation Impact:** This new API *definitely* requires documentation.  Where is it documented?
    * **Security Impact:** Could misuse of this API lead to security vulnerabilities (e.g., path traversal)?
    * **Compatibility Impact:** Does it break any existing code?

* **Insufficient Testing:** "has passed ostest" is inadequate.  Which specific ostest cases were run? What are the *actual* logs before and *after* the change?  Just saying "passed" doesn't provide any verifiable evidence.  Provide details about the build host (OS, compiler version) and the target (architecture, board, configuration).  Ideally, test on multiple configurations if possible.  Demonstrate that the function returns the expected directory name under different scenarios (e.g., root directory, subdirectories, relative paths).


**Example of an improved summary:**

"This PR implements the `get_current_dir_name()` API within the VFS layer to provide applications with a way to retrieve the name of the current working directory.  Currently, applications have no direct way to obtain this information, requiring workarounds. This API addresses this deficiency by providing a simple function call.  The implementation retrieves the current working directory path and extracts the last component (directory name) using path parsing functions.  This addresses Issue #[NuttX Issue Number]."


By providing more comprehensive information, the reviewer can effectively assess the change and ensure its quality and suitability for inclusion in NuttX.

File: libs/libc/unistd/lib_getcwd.c [hunk 1]
```diff
@@ -127,3 +127,42 @@ FAR char *getcwd(FAR char *buf, size_t size)
...
+ *   hold the absolute pathname of the current working directory.  If the
+ *   environment variable PWD is set, and its value is correct, then
+ *   that value will be returned. The caller should free the returned
+ *   buffer.
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   On success, get_current_dir_nametcwd() returns a pointer to a string
```
Comment (anchao) [inline libs/libc/unistd/lib_getcwd.c:145]:
```suggestion
 *   On success, get_current_dir_name() returns a pointer to a string
```

File: libs/libc/unistd/lib_getcwd.c [hunk 2]
```diff
@@ -127,3 +127,42 @@ FAR char *getcwd(FAR char *buf, size_t size)
...
+  FAR char *buf = lib_malloc(PATH_MAX + 1);
+  if (!buf)
+    {
+      set_errno(ENOMEM);
+      return NULL;
+    }
+
+  getcwd(buf, PATH_MAX);
+  return buf;
+}
```
Comment (anchao) [inline libs/libc/unistd/lib_getcwd.c:168]:
```suggestion
FAR char *get_current_dir_name(void)
{
  return getcwd(NULL, 0);
}
```

File: libs/libc/unistd/lib_getcwd.c [hunk 3]
```diff
@@ -127,3 +127,42 @@ FAR char *getcwd(FAR char *buf, size_t size)
...
+ *   On success, get_current_dir_name() returns a pointer to a string
+ *   containing the pathname of the current working directory.
+ *   Otherwise, get_current_dir_name() returns a null pointer and
+ *   sets errno to indicate the error:
+ *
+ *   EACCES
+ *     Read or search permission was denied for a component of the pathname.
+ *   ENOMEM
+ *     Insufficient storage space is available.
+ *
```
Comment (yamt) [inline libs/libc/unistd/lib_getcwd.c:154]: this seems to contain a copy-and-paste from linux man page. https://linux.die.net/man/3/get_current_dir_name
is it copyright-wise ok?

**Review by anchao — APPROVED** [review]

File: libs/libc/unistd/lib_getcwd.c (hunk 3 above)
Comment (extinguish) [reply libs/libc/unistd/lib_getcwd.c:154]: > this seems to contain a copy-and-paste from linux man page. https://linux.die.net/man/3/get_current_dir_name is it copyright-wise ok?

the comments is same as "getcwd(FAR char *buf, size_t size)", not copied from " https://linux.die.net/man/3/get_current_dir_name"

**Review by xiaoxiang781216 — APPROVED** [review]

File: libs/libc/unistd/lib_getcwd.c (hunk 3 above)
Comment (yamt) [reply libs/libc/unistd/lib_getcwd.c:150]: > > this seems to contain a copy-and-paste from linux man page. https://linux.die.net/man/3/get_current_dir_name is it copyright-wise ok?
> 
> the comments is same as "getcwd(FAR char *buf, size_t size)", not copied from " https://linux.die.net/man/3/get_current_dir_name"

sorry, what do you mean?
nuttx lib_getcwd.c doesn't seem to have the same comments.

**Review by yamt — CHANGES_REQUESTED** [review]

**Comment by yamt** [comment]
the description in this PR:
```
 *   get_current_dir_name() function will malloc an array big enough to
 *   hold the absolute pathname of the current working directory.  If the
 *   environment variable PWD is set, and its value is correct, then
 *   that value will be returned. The caller should free the returned
 *   buffer.
```

https://linux.die.net/man/3/get_current_dir_name:

get_current_dir_name() will [malloc](https://linux.die.net/man/3/malloc)(3) an array big enough to hold the absolute pathname of the current working directory. If the environment variable PWD is set, and its value is correct, then that value will be returned. The caller should [free](https://linux.die.net/man/3/free)(3) the returned buffer.

to me, they seem identical.
if you haven't copied the text from https://linux.die.net/man/3/get_current_dir_name,
can you make it clear from where you copied it?

**Comment by extinguish** [comment]
> the description in this PR:
> 
> ```
>  *   get_current_dir_name() function will malloc an array big enough to
>  *   hold the absolute pathname of the current working directory.  If the
>  *   environment variable PWD is set, and its value is correct, then
>  *   that value will be returned. The caller should free the returned
>  *   buffer.
> ```
> 
> https://linux.die.net/man/3/get_current_dir_name:
> 
> get_current_dir_name() will [malloc](https://linux.die.net/man/3/malloc)(3) an array big enough to hold the absolute pathname of the current working directory. If the environment variable PWD is set, and its value is correct, then that value will be returned. The caller should [free](https://linux.die.net/man/3/free)(3) the returned buffer.
> 
> to me, they seem identical. if you haven't copied the text from https://linux.die.net/man/3/get_current_dir_name, can you make it clear from where you copied it?

Sorry, I forgot where I referred to this comment from at that time. I've updated the comment information again to make sure it's different from the man page.  Please help review again.

**Pull request merged**